astropy/utils/tests/test_decorators.py

```diff
@@ -365,3 +365,60 @@ def _get_foo(cls): return 1
         foo = classproperty(_get_foo, doc="The foo.")
 
     assert B.__dict__['foo'].__doc__ == "The foo."
+
+
+def test_add_docstring_function():
+    docstring = 'test'
+
+    @add_docstring(docstring)
+    def testfunc():
+        pass
+
+    @add_docstring(docstring, replace=False)
+    def testfunc2():
+        """this is a """
+        pass
+
+    assert inspect.getdoc(testfunc) == docstring
```

Comment:
Probably over the top, but can you add one more test that shows that an existing docstring is in fact replaced?
With that, it is all OK as far as I'm concerned.